run-clang-tidy.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -e
+
+ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+[[ "$1" != "" && "$1" != "-fix" ]] && echo "The only supported argument is -fix" && exit
+
+# We are currently standardized on using LLVM/Clang10 for this script.
+# Note that this is totally independent of the version of LLVM that you
+# are using to build Halide itself. If you don't have LLVM10 installed,
+# you can usually install what you need easily via:
+#
+# sudo apt-get install llvm-10 clang-10 libclang-10-dev clang-tidy-10
+# export CLANG_TIDY_LLVM_INSTALL_DIR=/usr/lib/llvm-10
+
+[ -z "$CLANG_TIDY_LLVM_INSTALL_DIR" ] && echo "CLANG_TIDY_LLVM_INSTALL_DIR must point to an LLVM installation dir for this script." && exit
+echo CLANG_TIDY_LLVM_INSTALL_DIR = ${CLANG_TIDY_LLVM_INSTALL_DIR}
+
+# Use a temp folder for the CMake stuff here, so it's fresh & correct every time
+CLANG_TIDY_BUILD_DIR=`mktemp -d`
+echo CLANG_TIDY_BUILD_DIR = ${CLANG_TIDY_BUILD_DIR}
+
+echo Building compile_commands.json...
+cmake -DCMAKE_BUILD_TYPE=Debug \
+      -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+      -DLLVM_DIR=${CLANG_TIDY_LLVM_INSTALL_DIR} \
```

Comment:
I thought LLVM_DIR needs to be location of the cmake sub-folder, rather than the top-level installation dir?

Reply:
um... yeah, it does. How TF is this working?